packages/firestore/src/api/database.ts

```diff
@@ -495,9 +496,20 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
 
     const databaseInfo = this.makeDatabaseInfo();
 
+    this._firestoreClient = new FirestoreClient(
+      PlatformSupport.getPlatform(),
+      databaseInfo,
+      this._credentials,
+      this._queue
+    );
+
+    return this._firestoreClient.start(persistenceSettings);
+  }
+
+  private static makeDataConverter(databaseId: DatabaseId): UserDataConverter {
```

Comment:
I think I would like "createDataConverter" or "newDataConverter" slightly better. 

But... could we just make this a non-static function? As long as we return the UserDataConverter, we can still assign it in the constructor.

Reply:
SGTM.